scripts/nxutils.sh

```diff
@@ -17,38 +17,32 @@
 # limitations under the License.
 #
 # Contributors:
-#   Kevin Leturc
+#   Kevin Leturc, Julien Carsique
 #
 
+# Usage: "_retrieve_modules <result>"
+#   <result>: name of the variable which will receive the list of Maven modules of the current POM
 function _retrieve_modules {
-  echo $(mvn help:effective-pom -N|grep '<module>' |cut -d ">" -f 2 |cut -d "<" -f 1|sort|uniq|tr '\n' ' ')
+  local __resultvar=$1
+  local _list=LIST_$(echo $(basename $PWD)|tr '-' '_'|tr '[:lower:]' '[:upper:]')
+  if [ -z "${!_list}" ]; then
+    [ "$quiet" != true ] && echo "Modules list calculated from POM"
+    local _result="$(grep '<module>' pom.xml |cut -d ">" -f 2 |cut -d "<" -f 1|sort|uniq|tr '\n' ' ')"
+  else
+    [ "$quiet" != true ] && echo "Modules list set from environment variable: $_list"
+    local _result="${!_list}"
+  fi
+  eval "$__resultvar='$_result'"
```

Comment:
Not a hack but a commonly recommended way to set a local var by reference (from a function in this case); while using echo is a hack (or a bad practice in complex cases).
The called function must be able to echo text, debugging information and be resilient to wrong outputs.
Note that the "namerefs" is a built-in feature since BASH 4.3.